### DISCUSSION TITLE
Support extract timestamp with specified pattern in LAL.

### DISCUSSION THREAD

**Discussion by weixiang1862** [apache/skywalking#11484] [Ideas]
timestamp required a numeric string (millisecond) as parameter. In most case, we use patterned datetime in raw log for readability. If timestamp support parse time in specified pattern, it can cover all cases.
```
filter {
  json{
  }
  extractor{
    layer parsed.layer as String
    service parsed.service as String
    timestamp parsed.time as String                  <--------------------
    if (tag("LOG_KIND") == "SLOW_SQL") {
      slowSql {
        id parsed.id as String
        statement parsed.statement as String
        latency parsed.query_time as Long
      }
    }
  }
}
```
We can add a method `ExtractorSpec#timestamp(final String timestamp, final String formatPattern)`, then user can define time pattern freely in their raw log:
```
extractor {
    tag status: parsed.status
    tag remote_addr: parsed.remote_addr
    timestamp parsed.time as String, "yyyy-MM-dd'T'HH:mm:ssXXX"      <--------------------
}
```

**Comment by wu-sheng** [comment]
I think this is good to have. Are you going to submit a pull request for this?

cc @kezhenxu94

**Reply by weixiang1862** [reply]
Yes, I will submit after some tests.

**Comment by wu-sheng** [comment]
Provided by https://github.com/apache/skywalking/pull/11489
